src/core/agent/services/keriaNotificationService.ts

```diff
@@ -203,8 +209,48 @@ class KeriaNotificationService extends AgentService {
           // @TODO - Consider how to track/retry old notifications we couldn't process
           /* eslint-disable no-console */
           console.error("Error when process a notification", error);
+
+          const failedNotifications = await this.basicStorage.findById(
+            MiscRecordId.FAILED_NOTIFICATION
+          );
+
+          await this.basicStorage.createOrUpdateBasicRecord(
+            new BasicRecord({
+              id: MiscRecordId.FAILED_NOTIFICATION,
+              content: {
+                notifications: {
+                  ...(failedNotifications?.content.notifications || {}),
+                  [notif.i]: notif,
+                },
+                attempts: 1,
+                lastAttempt: Date.now(),
+              },
+            })
+          );
+
+          // Update the marker to not get stuck on this notification
+          notificationQuery.nextIndex += 1;
```

Comment:
I don't think we need to duplicate this. We can just put the try catch around only `await this.processNotification(notif);` and then update the marker in a common place after.